nomad/vault.go

```diff
@@ -1213,10 +1239,20 @@ func (v *vaultClient) EmitStats(period time.Duration, stopCh chan struct{}) {
 	for {
 		select {
 		case <-time.After(period):
-			stats := v.Stats()
+			stats := v.stats()
 			metrics.SetGauge([]string{"nomad", "vault", "distributed_tokens_revoking"}, float32(stats.TrackedForRevoke))
+			metrics.SetGauge([]string{"nomad", "vault", "token_ttl"}, float32(stats.TokenTTL/time.Millisecond))
 		case <-stopCh:
 			return
 		}
 	}
 }
+
+func tokenTTL(tokenData *tokenData) time.Duration {
+	if tokenData == nil {
+		return time.Duration(0)
+	}
+
+	ttl := int64(tokenData.CreationTime+tokenData.CreationTTL) - time.Now().Unix()
```

Comment:
The creation time isn't relevant to the TTL. You can keep renewing the token. We should be looking at the lastRenewed+lease_duration - time.Now()

https://www.vaultproject.io/api/auth/token/index.html#sample-response-5